konduit-serving-test/src/test/java/ai/konduit/serving/config/ConfigJsonCoverageTrackingTests.java

```diff
@@ -0,0 +1,271 @@
+        return m;
+    }
+
+    private static Method findStaticFromYaml(TextConfig c){
+        Method m = findStaticSingleStringArgMethodWithName(c.getClass(), "fromYaml");
+        Preconditions.checkState(m != null, "No fromJson(String) method is defined on class " + c.getClass().getName() +
+                " or any superclasses of this class. All classes implementing TextConfig should also have a static fromYaml(String) method");
+        return m;
+    }
+
+    private static Method findStaticSingleStringArgMethodWithName(Class<?> c, String methodName){
+        Class<?> current = c;
+        while(current != Object.class){
+            Method[] methods = current.getDeclaredMethods();
+            for(Method m : methods){
+                if(m.getName().equals(methodName) && m.getParameterCount() == 1 && Modifier.isStatic(m.getModifiers()) &&
+                        !Modifier.isAbstract(m.getModifiers()) && m.getParameterTypes()[0] == String.class){
+                    return m;
+                }
+            }
+            current = current.getSuperclass();
+        }
+        return null;
+    }
+
+    @Test
+    public void testInferenceConfiguration(){
+        InferenceConfiguration conf = InferenceConfiguration.builder()
+                .step(PythonStep.builder()
+                        .pythonConfig("x", PythonConfig.builder().pythonCode("import numpy as np\nreturn np.ones(1)").build())
+                        .inputNames(Arrays.asList("x"))
+                        .outputNames(Arrays.asList("z"))
+                        .build())
+                .step(ModelStep.builder().modelConfig(DL4JConfig.builder().modelConfigType(ModelConfigType.computationGraph("/my/model/path.bin")).build()).build())
+                .servingConfig(ServingConfig.builder().httpPort(12345).logTimings(true).build())
+                .build();
+
+//        System.out.println(conf.toYaml());
+
+        testConfigSerDe(conf);
+    }
+
+    @Test
+    public void testServingConfig(){
+        testConfigSerDe(ServingConfig.builder().httpPort(12345).logTimings(true)
+                .metricTypes(Arrays.asList(MetricType.JVM_GC, MetricType.CLASS_LOADER, MetricType.JVM_MEMORY, MetricType.JVM_THREAD))
+                .outputDataFormat(Output.DataFormat.JSON).build());
+    }
+
+    @Test
+    public void testArrayConcatenationStep(){
+        testConfigSerDe(ArrayConcatenationStep.builder().concatDimension(1,2).build());
+    }
+
+    @Test
+    public void testJsonExpanderTransformStep(){
+        testConfigSerDe(JsonExpanderTransformStep.builder()
+                .inputNames(Arrays.asList("x", "y"))
+                .outputNames(Arrays.asList("z","a"))
+                .inputSchema("x", new SchemaType[]{SchemaType.NDArray, SchemaType.Boolean})
+                .outputSchema("z", new SchemaType[]{SchemaType.Image, SchemaType.Categorical})
+        .build());
+    }
```

Comment:
There's some weird behavior with this test. The [toJson](https://github.com/KonduitAI/konduit-serving/pull/244/files#diff-fe2de6c522278bba41f95729aa38d514R94) method is adding a whitespace character at the end of `ai.konduit.serving.pipeline.step.JsonExpanderTransformStep<JsonExpanderTransformStep()>`

Reply:
`c.equals(fromJson);` is giving false for some reason. Is this because of `@EqualsAndHashCode(callSuper = true)`

Reply:
Looks like this is a bug/limitation of Java/Lombok, for arrays in generic collections - lombok just uses Object.equals, and so does HashMap.

I can reproduce it simply here:
https://gist.github.com/AlexDBlack/3a36cca09f30367023fe2e1a1e230760

So we need a custom equals method, easy enough fix.

Reply:
I ended up refactoring the use of arrays to lists, so we don't end up running into this sort if equality issue again anywhere else.